2-mandatory/4-stocks.js

```diff
@@ -63,8 +107,32 @@ function getPriceChanges(closingPricesForAllStocks) {
     The stock ticker should be capitalised.
     The price should be shown with exactly 2 decimal places.
 */
+function calculateHighestPrice(prices) {
+    let currentPrice = prices[0];
+
+    for(let i = 0; i < prices.length; i++) {
+        if(currentPrice < prices[i]) {
+            currentPrice = prices[i];
+        }
+    }
+
+    return currentPrice.toFixed(2);
+}
+
 function highestPriceDescriptions(closingPricesForAllStocks, stocks) {
-    // TODO
+    let priceDescription = [];
+
+    for(let i = 0; i < closingPricesForAllStocks.length; i++) {
+        let prices = closingPricesForAllStocks[i];
+        
+        let ticker = stocks[i].toUpperCase();
+        
+        let highestStockPrice = calculateHighestPrice(prices);
+        
+        priceDescription.push(`The highest price of ${ticker} in the last 5 days was ${highestStockPrice}`);
```

Comment:
Hi Hamra, great job!! The price should be shown with exactly 2 decimal places but I haven't seen in your code. Maybe you can use it after highestStockPrice.